File: src/main/kotlin/ru/vood/kotkin/rules/_03_expression/javaP/Try.java
package ru.vood.kotkin.rules._03_expression.javaP;

public class Try {
    public static void main(String[] args) {
        System.out.println("=== ДЕМО TRY ВЫРАЖЕНИЙ В JAVA ===\n");

        // ----------------------------------------------------------------------
        // 1. БАЗОВОЕ ИСПОЛЬЗОВАНИЕ TRY-CATCH КАК ВЫРАЖЕНИЯ
        // ----------------------------------------------------------------------
        System.out.println("1. БАЗОВОЕ ИСПОЛЬЗОВАНИЕ TRY-CATCH КАК ВЫРАЖЕНИЯ:");

//        Огромный минус: мутабельная переменная
        String result1;
        try {
            result1 = "Успешный результат: " + (10 / 2);
        } catch (ArithmeticException e) {
            result1 = "Ошибка вычисления: " + e.getMessage();
            // throw new RuntimeException("Asdsadasd");
            // result1 = null;
        }
        System.out.println("Результат: " + result1);
    }
}

File: src/main/kotlin/ru/vood/kotkin/rules/_03_expression/try.kt
package ru.vood.kotkin.rules.ru.vood.kotkin.rules._03_expression

fun main() {
    println("=== ДЕМО TRY ВЫРАЖЕНИЙ В KOTLIN ===\n")

    // ----------------------------------------------------------------------
    // 1. БАЗОВОЕ ИСПОЛЬЗОВАНИЕ TRY-CATCH КАК ВЫРАЖЕНИЯ
    // ----------------------------------------------------------------------
    println("1. БАЗОВОЕ ИСПОЛЬЗОВАНИЕ TRY-CATCH КАК ВЫРАЖЕНИЯ:")

    val result1: String = try {
        "Успешный результат: ${10 / 2}"
    } catch (e: ArithmeticException) {
        "Ошибка вычисления: ${e.message}"
//        throw RuntimeException("Asdsadasd")
//        null
    }
    println("Результат: $result1")
}